powerstore/resource_snapshotrule.go

```diff
@@ -0,0 +1,266 @@
+				MarkdownDescription: "Indicates whether this is a replica of a snapshot rule on a remote system.",
+			},
+			"nas_access_type": {
+				Type:                types.StringType,
+				Optional:            true,
+				Computed:            true,
+				Description:         "The NAS filesystem snapshot access method for snapshot rule.",
+				MarkdownDescription: "The NAS filesystem snapshot access method for snapshot rule.",
+			},
+			"is_read_only": {
+				Type:                types.BoolType,
+				Optional:            true,
+				Computed:            true,
+				Description:         "Indicates whether this snapshot rule can be modified.",
+				MarkdownDescription: "Indicates whether this snapshot rule can be modified.",
+			},
+			"managed_by": {
+				Type:                types.StringType,
+				Computed:            true,
+				Description:         "The entity that owns and manages the instance.",
+				MarkdownDescription: "The entity that owns and manages the instance.",
+			},
+			"managed_by_id": {
+				Type:                types.StringType,
+				Computed:            true,
+				Description:         "The unique id of the managing entity.",
+				MarkdownDescription: "The unique id of the managing entity.",
+			},
+		},
+	}, nil
+}
+
+// NewResource is a wrapper around provider
+func (r resourceSnapshotRuleType) NewResource(_ context.Context, p tfsdk.Provider) (tfsdk.Resource, diag.Diagnostics) {
+	return resourceSnapshot{
+		p: *(p.(*Pstoreprovider)),
+	}, nil
+}
+
+type resourceSnapshot struct {
+	p Pstoreprovider
+}
+
+func (r resourceSnapshot) Create(ctx context.Context, req tfsdk.CreateResourceRequest, resp *tfsdk.CreateResourceResponse) {
+	if !r.p.configured {
+		resp.Diagnostics.AddError(
+			"Provider not configured",
+			"The provider hasn't been configured before apply, likely because it depends on an unknown value from another resource. This leads to weird stuff happening, so we'd prefer if you didn't do that. Thanks!",
+		)
+		return
+	}
+
+	log.Printf("Started Creating Snapshot Rule")
+	var plan models.SnapshotRule
+
+	diags := req.Plan.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	snapshotRuleCreate := &gopowerstore.SnapshotRuleCreate{
+		Name:             plan.Name.Value,
+		Interval:         gopowerstore.SnapshotRuleIntervalEnum(plan.Interval.Value),
+		TimeOfDay:        plan.TimeOfDay.Value,
+		TimeZone:         gopowerstore.TimeZoneEnum(plan.TimeZone.Value),
+		DesiredRetention: int32(plan.DesiredRetention.Value),
+		NASAccessType:    gopowerstore.NASAccessTypeEnum(plan.NASAccessType.Value),
+		IsReadOnly:       plan.IsReadOnly.Value,
+	}
+
+	if len(plan.DaysOfWeek.Elems) > 0 {
+		snapshotRuleCreate.DaysOfWeek = []gopowerstore.DaysOfWeekEnum{}
+		for _, d := range plan.DaysOfWeek.Elems {
+			snapshotRuleCreate.DaysOfWeek = append(snapshotRuleCreate.DaysOfWeek, gopowerstore.DaysOfWeekEnum(d.String()))
+		}
+	}
+
+	// Add validation
```

Comment:
Please check if manual validation is required

Reply:
nope, server give back the error , so removing any type of validation for api